### PR TITLE
Filter data

### PR DESCRIPTION
**Pull request by maehr**
# Pull request

- Add: better JSON data via <https://forschung.stadtgeschichtebasel.ch/assets/data/metadata.json>
- Fix: metadata export inconsistencies
- Feat: reordering of metadata in item view
- Feat: improve readability of javascript snippets

## Types of changes

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Release Notes

- **New Features**
	- Enhanced data extraction and processing from the Omeka API, including support for CSV and JSON formats.
	- Improved pagination handling for item retrieval.
	- New functions for URL validation and combined value extraction.
	- Additional metadata fields for export and facets generation.
	- New upload step for JSON file in deployment workflow.

- **Improvements**
	- Streamlined metadata configuration for better clarity and expanded export fields.
	- Modernized JavaScript functions for improved performance and readability.
	- Simplified data access and improved URL generation logic in JSON exports.
	- Enhanced artifact handling in Jekyll deployment workflow.

- **Bug Fixes**
	- Enhanced error handling in data fetching processes and URL generation logic.

- **Chores**
	- New scripts added for cleaning and populating metadata.
	- Updated workflow for clearer artifact management during Jekyll site deployment.

### CODE REVIEW

File: .github/workflows/process_data.py [hunk 1]
```diff
@@ -1,110 +1,106 @@
...
 KEY_IDENTITY = os.getenv("KEY_IDENTITY")
 KEY_CREDENTIAL = os.getenv("KEY_CREDENTIAL")
 ITEM_SET_ID = os.getenv("ITEM_SET_ID")
+CSV_PATH = os.getenv("CSV_PATH", "_data/sgb-metadata-csv.csv")
+JSON_PATH = os.getenv("JSON_PATH", "_data/sgb-metadata-json.json")
+
+OMEKA_API_URL = "https://omeka.unibe.ch/api/"
+# OMEKA_API_URL = "https://omeka.unibe.ch/omeka-new/api/"
+KEY_IDENTITY = "sEJ4wdtCBpSkCzSPBpluzoqBzmkRCvw7"
+KEY_CREDENTIAL = "0v8AZu265B8EKkCCYOEkKQeOGBieejmy"
```
[the remark this answers is not in the document]
Comment (maehr) [reply .github/workflows/process_data.py]: What a stupid mistake from my side. I revoked all keys and deleted the hard coded parts.

**Review by koilebeit — CHANGES_REQUESTED** [review]
Thank you very much for your PR. Looks very good. But we should also catch the following cases that may occur:
- a parent object does not (yet) have any media objects (e.g. [https://omeka.unibe.ch/admin/item/10891](https://omeka.unibe.ch/admin/item/10891))
- the media object has no derivatives in OMEKA (i don't know the reason for this yet). See [https://omeka.unibe.ch/admin/item/11412](https://omeka.unibe.ch/admin/item/11412)

**Comment by maehr** [comment]
> Thank you very much for your PR. Looks very good. But we should also catch the following cases on the browse page that may occur:
> 
> * a parent object does not (yet) have any media objects (e.g. https://omeka.unibe.ch/admin/item/10891)
> * the media object has no derivatives in OMEKA (i don't know the reason for this yet). See https://omeka.unibe.ch/admin/item/11412

Thanks for pointing this out. I'll fix it.

The reason for the missing derivates are manual uploads because of the file size. We should report this to @flicksolutions directly.

**Review by koilebeit — COMMENTED** [review]
Now it works fine. But I actually liked the solution with the icon themes better. Now it always uses the same placeholder instead of the media-specific placeholder
https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/6fd1525ef348cf3819af2f9e7d4db4aee7bf5dc8/_includes/js/get-icon.js#L5
Now:
image
Before:
image

@mtwente, what do you think?

**Comment by maehr** [comment]
> Now it works fine. But I actually liked the solution with the icon themes better. Now it always uses the same placeholder instead of the media-specific placeholder
> 
> https://github.com/Stadt-Geschichte-Basel/forschung.stadtgeschichtebasel.ch/blob/6fd1525ef348cf3819af2f9e7d4db4aee7bf5dc8/_includes/js/get-icon.js#L5
> 
> 
> Now:
> image
> Before:
> image
> @mtwente, what do you think?

I'll remove the line for media objects, ok? If they are images they are not supposed to have no thumbs anyway

**Comment by maehr** [comment]
> > The reason for the missing derivates are manual uploads because of the file size. We should report this to @flicksolutions directly.
> 
> I just recreated all derivates on the new omeka server. The thumbnail generation fails these 4 tifs:
> 
> * 10722
> * 10723
>   -10953
>   -11413
> 
> I think the files might just be too large for the server to handle:
> 
> ```
> convert: width or height exceeds limit `/var/www/html/omeka-s-new/files/original/75d3863cda7afe2c253d7ff2c75b41b9f30e412f.tif' @ error/cache.c/OpenPixelCache/3911.
> convert: no images defined `/tmp/omekaOFSOEE.jpg' @ error/convert.c/ConvertImageCommand/3258.
> ```
> 
> Do you want me to look into that further?

Thank you very much for the swift reply. Can create the thumbnails manually and send them to you?

**Comment by flicksolutions** [comment]
> > > The reason for the missing derivates are manual uploads because of the file size. We should report this to @flicksolutions directly.
> > 
> > 
> > I just recreated all derivates on the new omeka server. The thumbnail generation fails these 4 tifs:
> > 
> > * 10722
> > * 10723
> >   -10953
> >   -11413
> > 
> > I think the files might just be too large for the server to handle:
> > ```
> > convert: width or height exceeds limit `/var/www/html/omeka-s-new/files/original/75d3863cda7afe2c253d7ff2c75b41b9f30e412f.tif' @ error/cache.c/OpenPixelCache/3911.
> > convert: no images defined `/tmp/omekaOFSOEE.jpg' @ error/convert.c/ConvertImageCommand/3258.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Do you want me to look into that further?
> 
> Thank you very much for the swift reply. Can create the thumbnails manually and send them to you?

That's worth a try. sure, go ahead

**Review by koilebeit — APPROVED** [review]

**Pull request merged**